webapp/channels/src/components/autocomplete_selector.tsx

```diff
@@ -1,29 +1,49 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
-    static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
-        onSelected: PropTypes.func,
-        label: PropTypes.node,
-        labelClassName: PropTypes.string,
-        inputClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string,
-        footer: PropTypes.node,
-        disabled: PropTypes.bool,
-        toggleFocus: PropTypes.func,
-        listComponent: PropTypes.elementType,
-        listPosition: PropTypes.string,
-    };
+import type ModalSuggestionList from './suggestion/modal_suggestion_list';
+import type Provider from './suggestion/provider';
+
+type Props = {
+    id?: string;
+    providers: Provider[];
+    value?: string;
```

Comment:
In theory, if defaultProps is defined, the props don't need to be optional. Typescript is smart enough to consider them optional on the outside, and "defined" in the inside.

So... you can set back all props in the default props to required.

(If this doesn't work as I said, let me know, and we can look into it)

Reply:
Yes, it is working 🙂. I usually work with Angular and it's a while since I last used React, so I was not aware of that. Today I learned 😄.